### ISSUE TITLE
bindCallback swallows error if it was thrown after the callback was executed, and the entire function is sycnhronous

### ISSUE THREAD

**Issue by olessavluk** [ReactiveX/rxjs#3487]
**RxJS version:**
5.5.8

**Code to reproduce:**
same for node and browser

```javascript
const fn = (datum, cb) => { 
  cb(datum);  
  throw new Error("callback error");
};

Rx.Observable.bindCallback(fn)(42)
  .subscribe(
    v => console.log(v), 
    e => console.error(e),
    c => console.log("completed"),
  );
```

**Expected behavior:**
see 42 and then "callback error" in the console

**Actual behavior:**
console output:
```
42
completed
```

**Additional information:**

In the real world, I've captured this issue with `ReactDOM.render` method. If you run it with "bad" JSX it will call passed callback first and then throw an error.

I've tried to reproduce this case as a test, but it worked well in the test environment. So maybe there is something with test scheduler or production build
<details>

```js
    it('should error if callback throws', () => {
      const expected = new Error('haha no callback for you');
      function callback(datum, cb) {
        cb(datum);
        throw expected;
      }
      const boundCallback = Observable.bindCallback(callback, null, rxTestScheduler);

      boundCallback(42)
        .subscribe((x: number) => {
          throw 'should not next';
        }, (err: any) => {
          expect(err).to.equal(expected);
        }, () => {
          throw 'should not complete';
        });

      rxTestScheduler.flush();
    });
```
</detail>

**Comment by benlesh** [comment]
I've updated the title of this issue, because it's specific to completely synchronous functions with callbacks.

**Comment by benlesh** [comment]
In the short term, the fix for this is to pass a scheduler to the bindCallback function:  `bindCallback(fn, asapScheduler)` for example.

**Comment by cartant** [comment]
Closing this because it's addressed in #4089.
